orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy

```diff
@@ -72,21 +72,29 @@ class MonitorKatoTask implements RetryableTask {
     }
 
     Task katoTask
+    def firstNotFoundRetrySkippingReplica = stage.context."kato.task.firstNotFoundRetry.skipReplica" as Long
     try {
-      katoTask = kato.lookupTask(taskId.id).toBlocking().first()
+      katoTask = kato.lookupTask(taskId.id, firstNotFoundRetrySkippingReplica != null).toBlocking().first()
     } catch (RetrofitError re) {
       //handle a 404 if a task update has not successfully replicated to a read replica
       if (re.kind == RetrofitError.Kind.HTTP && re.response.status == HttpURLConnection.HTTP_NOT_FOUND) {
-        def firstNotFoundRetry = stage.context."kato.task.firstNotFoundRetry" as Long
+        def firstNotFoundRetry = !firstNotFoundRetrySkippingReplica ? stage.context."kato.task.firstNotFoundRetry" as Long
+          : firstNotFoundRetrySkippingReplica
+
         def now = clock.millis()
         def ctx = [:]
         if (firstNotFoundRetry == null || firstNotFoundRetry == -1) {
           ctx['kato.task.firstNotFoundRetry'] = now
           firstNotFoundRetry = now
         }
+
         if (now - firstNotFoundRetry > TASK_NOT_FOUND_TIMEOUT) {
+          if (firstNotFoundRetrySkippingReplica) {
```

Comment:
Is this just an optimization? Are you thinking you want it to immediately fail the first time it gets a 404 directly from the master?

If so maybe a comment along those lines. I also think the variable being called `skipReplica` is a little clearer than `firstNotFoundRetrySkippingReplica`.